src/platform/ESP32/NetworkCommissioningDriver.cpp

```diff
@@ -203,21 +222,34 @@ void ESPWiFiDriver::OnConnectWiFiNetworkFailed(chip::System::Layer * aLayer, voi
 
 void ESPWiFiDriver::ConnectNetwork(ByteSpan networkId, ConnectCallback * callback)
 {
-    CHIP_ERROR err              = CHIP_NO_ERROR;
-    Status networkingStatus     = Status::kSuccess;
+    CHIP_ERROR err          = CHIP_NO_ERROR;
+    Status networkingStatus = Status::kSuccess;
+    Network configuredNetwork;
     const uint32_t secToMiliSec = 1000;
 
     VerifyOrExit(NetworkMatch(mStagingNetwork, networkId), networkingStatus = Status::kNetworkIDNotFound);
     VerifyOrExit(mpConnectCallback == nullptr, networkingStatus = Status::kUnknownError);
     ChipLogProgress(NetworkProvisioning, "ESP NetworkCommissioningDelegate: SSID: %.*s", static_cast<int>(networkId.size()),
                     networkId.data());
-
+    if (CHIP_NO_ERROR == GetConfiguredNetwork(configuredNetwork))
+    {
+        if (configuredNetwork.networkIDLen == mStagingNetwork.ssidLen &&
+            memcmp(configuredNetwork.networkID, mStagingNetwork.ssid, mStagingNetwork.ssidLen) == 0)
```

Comment:
Instead of rewriting you can use [`NetworkMatch()`](https://github.com/project-chip/connectedhomeip/blob/master/src/platform/ESP32/NetworkCommissioningDriver.cpp#L89-L92) API, it basically does the same thing.

Reply:
Done